README.md

```diff
@@ -64,6 +68,195 @@ org.hwangsaeul.hwangsae.recorder relay-stream-port uint32 9999
 For more details check the file
 `hwangsae/org.hwangsaeul.hwangsae.gschema.xml`
 
+## Broker API
```

Comment:
Hmm, these out-of-source API documentations tend to be neglected and will quickly get out of sync with the actual code. Couldn't we rather document the API right in the respective header files in some machine-readable format and generate the documentation out of it?

[gtk-doc](https://developer.gnome.org/gtk-doc-manual/stable/) comes to mind. `glib` has many [examples](https://gitlab.gnome.org/GNOME/glib/blob/master/gobject/gobject.h) how to use it. It is also integrated with [meson](https://mesonbuild.com/Gnome-module.html#gnomegtkdoc).

Reply:
I see your point. Thanks for your suggestion.

Reply:
@xhaakon , @justinjoy I would like to confirm if my understanding is correct. The initial approach was to use Github `MD` files to render the documentation, and use links to connect the different modules. However if we use  `gtk-doc` we would need to parse and install the output files to some web server to be able to access them online.
Is this correct? Should this documentation be online? 
In case the docs should be online, should we publish the output to the project site? Any other approach?

Reply:
I think we should generate the apidocs and push them to some repository on GitHub to which you can point from the main documentation.

I guess we could automate this with CI e.g. on every merge.

Reply:
Yes, we can use CI, but please, do not focus on CI. :)

Reply:
OK.